instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/JarFileDetector.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources;
+
+import static java.util.logging.Level.WARNING;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.jar.Manifest;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+class JarFileDetector {
```

Comment:
`SpringBootServiceNameDetector `and  `JarServiceNameDetector ` implement `ConditionalResourceProvider`, which is not the case for `JarFileDetector`. So, the name `JarFileDetector` is a bit confusing.

Reply:
Right - will rename to `JarPathFinder`